File: Turnup/Turnup/Turnup/Tests/TM_Tests.cs
using Login.Pages;
using OpenQA.Selenium.Chrome;
using OpenQA.Selenium;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Turnup.Utilis;
using NUnit.Framework;

namespace Turnup.Tests
{
    //[Parallelizable]
    [TestFixture]
    public class TM_Tests : CommonDriver
    {

        LoginPage loginPageObj = new LoginPage();
        HomePage HomePageObj = new HomePage();
        TMPage TMPageObj = new TMPage();
        [SetUp]
        public void TimeSetup()
        {

            IWebDriver dr = new ChromeDriver();
            //Login page object initialization and deifinition
            loginPageObj.LoginActions(dr);

            // Home page object initialization and deifinition

            HomePageObj.GoToTMPage(dr);
        }

            [Test, Order(1), Description("This test is creating a new Time record")]
            public void CreateTime_Test()
            {
                // TM Page object initialization and definition

                TMPageObj.CreateNewTime(dr);

            }
            [Test, Order(2), Description("This test is creating a new Time record")]
            public void EditTime_Test()
            {
                TMPageObj.EditTMPage(dr);
            }
            [Test, Order(3), Description("This test is deleting an existing Time record")]
            public void DeleteTime_Test()
            {
                TMPageObj.DeleteTMPage(dr);
            }

            [TearDown]
            public void CloseTestRun()
            {
                dr.Quit();
            }

        }
    }

File: src/test/java/Tests/TM_Tests.java
package Tests;

import Pages.HomePage;
import Pages.LoginPage;
import Pages.TM_Page;
import Utilies.CommonDriver;
import org.junit.jupiter.api.Test;
import org.openqa.selenium.chrome.ChromeDriver;
import org.testng.annotations.AfterTest;
import org.testng.annotations.BeforeTest;

public class TM_Tests extends CommonDriver {


        LoginPage loginPageObj = new LoginPage();
        HomePage homePageObj = new HomePage();
        TM_Page tmPageObj = new TM_Page();

        @BeforeTest
        public void TM_SetUp() throws InterruptedException {
            // open Chrome browser
            dr = new ChromeDriver();
            Thread.sleep(1000);

            // Login page object initialization and definition

            loginPageObj.LoginActions(dr);

            // Home page object intialization and definition

            homePageObj.GoToTMPage(dr);
        }

        @Test(priority=0, description = " This test checks if a user is able to create a new time record")
        public void CreateTime_Test() throws InterruptedException {
            // TM page object initialization and definition

            tmPageObj.CreateNewTime(dr, "Keyboard", "M","Unknown Material","500");
            tmPageObj.CreateTMAssertion(dr, "Keyboard", "M","Unknown Material","$500.00");
        }

        @Test(priority = 1, description = "This test checks if a user is able to edit an existing time record")
        public void EditTime_Test()
        {
            tmPageObj.EditTMPage(dr, "Mouse", "M","Known Material","100");
            tmPageObj.EditTMPage(dr,"Mouse", "M","Known Material","100");
        }

        @Test(priority = 2, description = "This test checks if a user is able to delete an existing time record")
        public void DeleteTime_Test() throws InterruptedException {
            tmPageObj.DeleteTMPage(dr);
            tmPageObj.DeleteTMAssertion(dr,"M","Know Material","100");
        }

        @AfterTest
        public void ClosingSteps()
        {
            dr.quit();
        }
    }

